.gitignore

```diff
@@ -119,23 +119,19 @@ nncf_debug/
 
 # NNCF examples
 examples/torch/object_detection/eval/
-examples/post_training_quantization/onnx/mobilenet_v2/mobilenet_v2_*
-examples/post_training_quantization/openvino/mobilenet_v2/mobilenet_v2_*
-examples/post_training_quantization/tensorflow/mobilenet_v2/mobilenet_v2_*
-examples/post_training_quantization/torch/mobilenet_v2/mobilenet_v2_*
-examples/post_training_quantization/torch/ssd300_vgg16/ssd300_vgg16_*
-examples/post_training_quantization/openvino/anomaly_stfpm_quantize_with_accuracy_control/stfpm_*
-examples/post_training_quantization/openvino/yolov8/yolov8n*
-examples/post_training_quantization/openvino/yolov8_quantize_with_accuracy_control/yolov8n*
-examples/**/runs/**
-examples/**/results/**
-examples/llm_compression/openvino/tiny_llama_find_hyperparams/statistics
-compressed_graph.dot
-original_graph.dot
-datasets/**
```

Comment:
`datasets` directory created by ultralitics, it's only happens if you have yolo config `.config/Ultralytics/settings.json` file that was generated some time ago or changes manually.

On current version of yolo, default path for `datasets_dir` generated as `nncf/../datasest` if run example from NNCF in case of  `.config/Ultralytics/settings.json`  is not exist.
If datasets directory created on your host you need to modify config by `yolo settings dataset_dir=new_path` or by text editor, or remove config file after it will be generated this new path.

Reply:
I do not agree.
I've run the following command from the `nncf` root (as usual, for testing purposes):
`pytest -s tests/cross_fw/examples/test_examples.py -k "post_training_quantization_openvino_yolo8_quantize"`
And after the passing, I observe the following directory:
![image](https://github.com/user-attachments/assets/79a328c3-74ff-43fe-b66c-103563aa14e1)
Thus, removing `datasets/**` from the `.gitignore` might cause issues.

Reply:
From a high perspective, it would be more helpful if you would update examples to create a `datasets` folder during the run in a corresponding or temporary directory, to exclude incidents in the future. In this case, we would be able to remove this path from `.gitignore` and safely run tests.

Reply:
Have you updated the ultralytics config file as described above?

Reply:
You have config file `~/.config/Ultralytics/settings.json` on your host that was generated when you run any code with ultralitics for the first time and creating or removing any virtual environment does not affect on the config in any way.

Your config collect line like `"datasets_dir": "/home/<username>/nncf/datasets",`.
This line only generated if run NNCF examples with old version of ultralitics, current default value is like `/home/<username>/datasets` - `GIT_ROOT.parent / "datasets"`.

There is only one way conditions to get `datasets_dir` like `.../nncf/datasets`:
- You did not run any code with ultralitics before
- And for the first time you have run NNCF yolo example with old version of ultralitics

In all other cases `datasets_dir` will not set as `nncf/datasets`

My point, If you have config file of any third party package with not default values in your `~/.config` directory, that used NNCF folder as storage for anything, it's not responsibility of NNCF, it's incorrect configuration that should be resolved by user.

**Moreover**, the `datasets` directory in the NNCF git root folder breaks `isort`, by changing the import order of `datasets` module. 
So `datasets` in `.gitignore` hides this problem by making the existence of this directory expected but it is not.

@alexsu52 as you have added `datasets` to `.gitignore`, i think last word is your

Reply:
> it's not responsibility of NNCF, it's incorrect configuration that should be resolved by user.

If I run NNCF tests from the root and NNCF creates a `datasets` folder in the root, this is NNCF's responsibility, not the user, since it is not possible to set up the folder for this via tests.
As you said, 

> datasets directory in the NNCF git root folder breaks isort

So, this is another reason to change the placing of the `datasets` folder on the NNCF's side, since tests were designed by the NNCF team.

Reply:
I read your discussion and did the following experiment:
- remove `~/.config/Ultralytics/settings.json`
- create a clear env using
```
pip install -r examples/post_training_quantization/onnx/yolov8_quantize_with_accuracy_control/requirements.txt
```

Ultralytics version:
![image](https://github.com/user-attachments/assets/4e909c15-1c08-41a4-8d55-3f6db5d5787d)

- run example
```
python examples/post_training_quantization/onnx/yolov8_quantize_with_accuracy_control/main.py
```
dataset was downloaded into `nncf/datasets`
![image](https://github.com/user-attachments/assets/cbf9f43d-8c86-408c-be17-62ef56ebe813)

Where did I go wrong?